docs/jetstream/simplified.md

```diff
@@ -0,0 +1,498 @@
+at a pace the client can sustain and how the reading happens.
+
+The JetStream API supports three strategies for retrieving messages:
+- request the next message, i.e., one message at a time
+- request the next N messages
+- continuously request and maintain a buffer of N messages.
+
+The first two options allow the client to control and manage its buffering manually.
+When the client is done processing the messages, it can, at its discretion,
+request additional messages or not.
+
+The last option auto-buffers messages for the client, thus controlling the message and data rates.
+The client specifies how many messages it wants to receive, and as it consumes them, the pull requests
+additional messages in an effort to prevent the consumer from stalling and thus maximize performance.
+
+### Message
+
+All messages received through a consumer are of `NATS::JetStream::Message`. It has all the standard
+fields of a NATS `Msg` and provides functionality for inspecting metadata encoded into the message's
+reply subject. This metadata includes:
+
+- `sequence` or `seq` - the sequence information for the message
+- `num_delivered` - the number of times this message was delivered to the consumer
+- `num_pending` - the number of messages that match the consumer's filter but have not been delivered yet
+- `timestamp` - the time the message was originally stored on a stream
+- `stream` - the stream name this message is stored on
+- `consumer` - the consumer name this message was delivered to
+- `domain` - the domain this message was received on.
+
+### Next
+
+The simplest mechanism to process messages is to request a single message with `next`:
+
+```ruby
+message = consumer.next(options)
+```
+
+It will block until the message is received or the request has expired. When no messages are available,
+or an error occurs during the retrieval, the request will return `nil`.
+
+The method takes two optional arguments:
+- `:expires` - amount of time to wait for the request to expire (in nanoseconds). Minimum is 1 second, defaults to 30 seconds.
+- `:idle_heartbeat` - amount idle time the server should wait before sending a heartbeat. Minimum is 500ms, maximum is 30 seconds. Defaults to half of `expires`.
+
+### Fetch
+
+You can request multiple messages at a time with `fetch`:
+
+```ruby
+messages = consumer.fetch(options)
+
+messages.each do |message|
+  puts message
+end
+```
+
+The request is a long poll, and it will block until the desired number of messages is received or the expires time triggers.
+This means that the number of messages you request is only a hint, and it is just the upper bound on the number of messages you will receive.
+
+By default, `fetch` will retrieve 100 messages in a batch, but you can control how many messages you receive with options:
+
+- `:max_messages` - max number of messages to return, cannot be combined with `max_bytes`
+- `:max_bytes` - max number of bytes to return, cannot be combined with `max_messages`
```

Comment:
max messages and max bytes being exclusive was an oversight at the time, it should be possible to set them both as in https://github.com/nats-io/nats.go/pull/1789